File: src/sdk-client/Microsoft.Windows.Midi.Minimal/MidiEndpoint.cpp
// ------------------------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the GitHub project root for license information.
// ------------------------------------------------------------------------------------------------

#define WINDOWSMIDISERVICES_EXPORTS


#include "WindowsMidiServicesClient.h"

namespace Microsoft::Windows::Midi
{
	struct MidiEndpoint::impl
	{
		Enumeration::MidiEndpointInformation Information;
		MidiMessagesReceivedCallback _messagesReceivedCallback;

		//MidiEndpoint(const Enumeration::MidiEndpointInformation& information, const MidiMessagesReceivedCallback& messagesReceivedCallback);
		//MidiEndpoint(const Enumeration::MidiEndpointInformation& information);

		// TODO: Vector of groups / channels / protocol versions / other MIDI CI information
	};

}

File: src/sdk-client/Windows.Midi/Devices/MidiEndpoint.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using Microsoft.Windows.Midi.Enumeration;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Microsoft.Windows.Midi.Devices
{
    public sealed class MidiEndpoint
    {
        public MidiEndpointInformation EndpointInformation { get; internal set; }


        public void Close()
        {
        }

    }
}
